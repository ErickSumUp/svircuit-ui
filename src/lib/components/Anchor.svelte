<script lang="ts">
  import type { HTMLAnchorAttributes } from 'svelte/elements';

  interface Props extends HTMLAnchorAttributes {
    size?: 'one' | 'two';
    variant?: '' | 'highlight' | 'quote';
  }

  let { size = 'one', variant = '', href, children, ...rest }: Props = $props();
</script>

<a
  {href}
  class="base"
  class:one={size === 'one'}
  class:two={size === 'two'}
  class:highlight={variant === 'highlight'}
  class:quote={variant === 'quote'}
  {...rest}
>
  {@render children()}</a
>

<style>
  .base {
    display: inline-block;
    padding: 0;
    margin-top: 0;
    margin-right: 0;
    margin-left: 0;
    color: var(--cui-fg-accent);
    text-decoration: underline;
    text-decoration-skip-ink: auto;
    background: none;
    border: 0;
    border-radius: var(--cui-border-radius-byte);
    outline: none;
    transition:
      opacity var(--cui-transitions-default),
      color var(--cui-transitions-default),
      background-color var(--cui-transitions-default),
      border-color var(--cui-transitions-default);
  }

  .base:hover {
    color: var(--cui-fg-accent-hovered);
    cursor: pointer;
  }

  .base:active {
    color: var(--cui-fg-accent-pressed);
  }

  .base {
    font-weight: var(--cui-font-weight-regular);
  }
  .one {
    font-size: var(--cui-typography-body-s-font-size);
    line-height: var(--cui-typography-body-s-line-height);
  }
  .two {
    font-size: var(--cui-typography-body-m-font-size);
    line-height: var(--cui-typography-body-m-line-height);
  }

  .highlight {
    font-weight: var(--cui-font-weight-bold);
  }

  .quote {
    font-style: italic;
  }
</style>
