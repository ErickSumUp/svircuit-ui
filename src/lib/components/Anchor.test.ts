import { render } from '@testing-library/svelte';
import '@testing-library/jest-dom';
import Anchor from '$lib/components/Anchor.svelte';
import { test, expect } from 'vitest';

test('should match story case', async () => {
  const { container } = render(Anchor, {
    props: { href: 'https://opensource.sumup.com/' }
  });
  expect(container).toMatchSnapshot();
});
