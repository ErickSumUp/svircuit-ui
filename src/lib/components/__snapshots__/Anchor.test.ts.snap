// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match story case 1`] = `
<div>
  <a
    class="base svelte-18jryi6 one"
    href="https://opensource.sumup.com/"
  >
    Example Snippet
    <!---->
  </a>
  
</div>
`;
